### DISCUSSION TITLE
Why the getter and setter methods have not been triggered?

### DISCUSSION THREAD

**Discussion by yohowie** [phalcon/cphalcon#15715] [Q&A]
Users.php model code
```
protected $status;

public function setStatus($status)
    {
        $this->status = $status;
    }
    
    public function getStatus()
    {
        return $this->status = 'test';
    }
    
    public function afterFetch()
    {
        $this->status = 'test_1';
    }
```
UserController.php controller code
```
$user = Users::findFirst([
    'conditions' => 'id = 2',
]);
var_dump($user->toArray());
```
[Status] field can be replaced for 'test_1'

If you query the specified field, the [status] field will not be modified
```
$user = Users::findFirst([
    'conditions' => 'id = 2',
    'columns' => 'name, status'
]);
```
Why the getter and setter methods have not been triggered?
Why the afterFetch method cannot be triggered normally when the query field is specified?

**Comment by quasipickle** [comment]
Can you explain more what happens when you say this: `[Status] field can be replaced for 'test_1'`?  It's not clear to me what you're doing that makes you think `setStatus()` should be called.

**Reply by yohowie** [reply]
My idea is to use the model to query the data, automatically call the getStatus method when the data is returned, and modify the result of the status field.
